### DISCUSSION TITLE
i new using the axmol

### DISCUSSION THREAD

**Discussion by PedroQuesado** [axmolengine/axmol#2177] [General]
Hello all
how i implement the farygui in my project 
i want change from gui to farygui
can you help-me
my project in made in c++

**Comment by rh101** [comment]
Check the `{axmol}/tests/fairygui-tests` project.  For any specific documentation to FairyGUI, you would need to check the FairyGUI [website](https://www.fairygui.com/)

**Comment by PedroQuesado** [comment]
i using the version 2.1.5
i only have to put this in cmake
 `if(NOT DEFINED BUILD_ENGINE_DONE)
    set(AX_ENABLE_EXT_FAIRYGUI ON CACHE BOOL "Build extension FairyGUI" FORCE)
    set(AX_ENABLE_EXT_SPINE ON CACHE BOOL "Build extension spine" FORCE)
    set(AX_ENABLE_EXT_DRAGONBONES OFF CACHE BOOL "Build extension DragonBones" FORCE)
    set(AX_ENABLE_EXT_COCOSTUDIO OFF CACHE BOOL "Build extension cocostudio" FORCE)
    set(AX_ENABLE_EXT_ASSETMANAGER OFF CACHE BOOL "Build extension asset-manager" FORCE)
    set(AX_ENABLE_EXT_PARTICLE3D OFF CACHE BOOL "Build extension Particle3D" FORCE)
    set(AX_ENABLE_EXT_LUA OFF CACHE BOOL "Build lua libraries" FORCE)
    set(AX_ENABLE_EXT_GUI OFF CACHE BOOL "Build extension GUI" FORCE)
    set(AX_ENABLE_EXT_JSONDEFAULT OFF CACHE BOOL "Build extension JSONDefault" FORCE)
    set(AX_ENABLE_EXT_PHYSICS_NODE OFF CACHE BOOL "Build extension physics-nodes" FORCE)`  
and after i have to declare the libray in the code for example #include "faryGui.h" 
i have this error
 `Gravidade	Código	Descrição	Projeto	Arquivo	Linha	Estado de Supressão	Detalhes
Erro	MSB8066	o build personalizado para 'C:\CMakeLists.txt' foi encerrado com o código 1.	RuinBound	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	254		
Erro	MSB8066	o build personalizado para 'C:\uinBound\build\CMakeFiles\637762351dea45a76913826f548b6376\SYNC_RESOURCE-RuinBound.rule; C:\RuinBound\CMakeLists.txt' foi encerrado com o código 1.	SYNC_RESOURCE-RuinBound	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	254		
Erro	MSB8066	o build personalizado para 'C:\RuinBound\build\CMakeFiles\637762351dea45a76913826f548b6376\SYNC_RESOURCE-RuinBound-axslc.rule;C:\RuinBound\CMakeLists.txt' foi encerrado com o código 1.	SYNC_RESOURCE-RuinBound-axslc	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	254		
Erro	MSB8066	o build personalizado para 'C:\RuinBound\build\CMakeFiles\637762351dea45a76913826f548b6376\generate.stamp.rule' foi encerrado com o código 1.	ZERO_CHECK	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	237		
`

**Reply by rh101** [reply]
It's extremely hard to read the text you pasted, since it's not formatted in any way.  In future, please ensure that the line-breaks are there, and add the backtick character before and after code or log sections that you want to paste.  For single lines use a single backtick, for multiple lines surround them with 3 backticks before and after, or use the button marked in red in this screenshot:

![image](https://github.com/user-attachments/assets/65c9dba2-c6a1-4b0b-a875-f6955aebf917)

which results in this:
```
if(NOT DEFINED BUILD_ENGINE_DONE) 
  set(AX_ENABLE_EXT_FAIRYGUI ON CACHE BOOL "Build extension FairyGUI" FORCE) 
  set(AX_ENABLE_EXT_SPINE ON CACHE BOOL "Build extension spine" FORCE) 
  set(AX_ENABLE_EXT_DRAGONBONES OFF CACHE BOOL "Build extension DragonBones" FORCE) 
  set(AX_ENABLE_EXT_COCOSTUDIO OFF CACHE BOOL "Build extension cocostudio" FORCE) 
  set(AX_ENABLE_EXT_ASSETMANAGER OFF CACHE BOOL "Build extension asset-manager" FORCE) 
  set(AX_ENABLE_EXT_PARTICLE3D OFF CACHE BOOL "Build extension Particle3D" FORCE) 
  set(AX_ENABLE_EXT_LUA OFF CACHE BOOL "Build lua libraries" FORCE) 
  set(AX_ENABLE_EXT_GUI OFF CACHE BOOL "Build extension GUI" FORCE) 
  set(AX_ENABLE_EXT_JSONDEFAULT OFF CACHE BOOL "Build extension JSONDefault" FORCE) 
  set(AX_ENABLE_EXT_PHYSICS_NODE OFF CACHE BOOL "Build extension physics-nodes" FORCE)
```


Now, did you try to compile the project before adding the `#include "faryGui.h"`?  Also, not sure if you had a typo just in your post, but it should be `#include "FairyGUI.h"`.

Make sure your project builds correctly without the `#include "FairyGUI.h"` line.

**Reply by PedroQuesado** [reply]
![image](https://github.com/user-attachments/assets/da9780f6-e316-49cd-8c68-fb935c157a35)
![image](https://github.com/user-attachments/assets/96b731d3-32ea-4b63-804b-9ae8872696b4)
![image](https://github.com/user-attachments/assets/d4e63276-5d3d-4980-9960-467f75c3f46f)

 i put some print of my cmake 
 i remove the library
if don´t use the cmake like its in the print e compile correctly and don´t give any error if i put #include "farygui.h"
i put the link to my cmake
https://github.com/PedroQuesado/axmolcmake/blob/main/cmke

**Reply by rh101** [reply]
Something isn't quite right with your project, based on what I can see in those screenshots.  In one screenshot there is this line:
`_1klink("${sample-assets_SOURCE_DIR}/cpp-tests/Content" "${CMAKE_CURRENT_LIST_DIR}/Content")`

That is not part of the `CMakeLists.txt` template, and it looks like it is a modified cpp-tests project.

Did you follow the [DevSetup](https://github.com/axmolengine/axmol/blob/dev/docs/DevSetup.md) guide on generating a new axmol project?  Please show the exact commands you used to generate the project and build files.  

If you didn't follow the DevSetup guide, then please do so.  Once you generate a new project and build files, then build that project, and make sure it works, before making any changes to it.

**Comment by PedroQuesado** [comment]
i uses this command 
`axmol new -p org.axmol.RuinBound  -l cpp --portrait RuinBound
`
next i use this command
`cmake -S . -B build -G "Visual Studio 17 2022" -A Win32`
removing this line
`_1klink("${sample-assets_SOURCE_DIR}/cpp-tests/Content" "${CMAKE_CURRENT_LIST_DIR}/Content")`
i have the same error
if i remove 
`if(NOT DEFINED BUILD_ENGINE_DONE) `
 ` set(AX_ENABLE_EXT_FAIRYGUI ON CACHE BOOL "Build extension FairyGUI" FORCE) `
 ` set(AX_ENABLE_EXT_SPINE ON CACHE BOOL "Build extension spine" FORCE) `
`  set(AX_ENABLE_EXT_DRAGONBONES OFF CACHE BOOL "Build extension DragonBones" FORCE) `
`  set(AX_ENABLE_EXT_COCOSTUDIO OFF CACHE BOOL "Build extension cocostudio" FORCE) `
`  set(AX_ENABLE_EXT_ASSETMANAGER OFF CACHE BOOL "Build extension asset-manager" FORCE) `
`  set(AX_ENABLE_EXT_PARTICLE3D OFF CACHE BOOL "Build extension Particle3D" FORCE) `
`  set(AX_ENABLE_EXT_LUA OFF CACHE BOOL "Build lua libraries" FORCE) `
`  set(AX_ENABLE_EXT_GUI OFF CACHE BOOL "Build extension GUI" FORCE) `
`  set(AX_ENABLE_EXT_JSONDEFAULT OFF CACHE BOOL "Build extension JSONDefault" FORCE) `
`  set(AX_ENABLE_EXT_PHYSICS_NODE OFF CACHE BOOL "Build extension physics-nodes" FORCE)` my code compile correctly

**Reply by rh101** [reply]
Please follow these instructions, and then reply back with whether the project works or not:

1. `axmol new -p org.axmol.RuinBound -l cpp --portrait RuinBound `
2. `cd RuinBound`
3. `axmol build -p win32 -c` or `axmol -p win32 -c`  (the `build` keyword is now optional)
4. Open the `RuinBound/build/RuinBound.sln` solution in Visual Studio 2022
5. Build the project.  Ensure that it builds and runs successfully.

Once this is working, only then make modifications to the `RuinBound/CMakeLists.txt`, but only adding the following:

```
set(AX_ENABLE_EXT_FAIRYGUI ON CACHE BOOL "Build extension FairyGUI" FORCE)
set(AX_ENABLE_EXT_SPINE ON CACHE BOOL "Build extension spine" FORCE)
```

Then re-run `axmol -p win32 -c -f`.   After this,  ensure that the `AX_ENABLE_EXT_FAIRYGUI` and `AX_ENABLE_EXT_SPINE` cached values are actually updated.  You can check for this by opening up `RuinBound/build/CMakeCache.txt`, and searching for `AX_ENABLE_EXT_FAIRYGUI`.  It must be `ON`.  If it is not, then delete the `CMakeCache.txt`, and re-run the `axmol -p win32 -c -f` command.

Do not add any other cache settings (`set(AX_ENABLE_EXT_*****`) until you have tested with just the two listed above.

**Reply by PedroQuesado** [reply]
its work
can i replicate that for my current code or i have to create a new project
its possible to force create a project in the directory already exist
because i want put the code in the git
for now and creating in other path and move all project to directory of my git project

**Reply by rh101** [reply]
You don't need to create a new project.  Compare the files from the working project to your project, and make the necessary changes.  

The fact that a newly generated Axmol project works means that you need to review your own project to figure out what is causing the issues.  The only other suggestions I can make is that you remove or comment out anything unnecessary from your project, and then add one item at a time and test the build, and continue doing that until you figure out which is causing the issue.

**Reply by smilediver** [reply]
I think the best way to go about this is to:

* Move your old original (Cocos) `CMakeLists.txt` to `CMakeLists.txt.old`
* Generate a new Axmol project with `axmol new ...`
* Move the `CMakeLists.txt` from generated project to your project
* Then go line by line in `CMakeLists.txt.old` and see if you need to transfer the customizations to your new `CMakeLists.txt`
* After you're done you can delete `CMakeLists.txt.old`

**Reply by PedroQuesado** [reply]
its works
i make the changes directly in my project and its working
thanks for your help

**Reply by PedroQuesado** [reply]
to use the fary gui only need this 
`set(AX_ENABLE_EXT_FAIRYGUI ON CACHE BOOL "Build extension FairyGUI" FORCE)`
`set(AX_ENABLE_EXT_SPINE ON CACHE BOOL "Build extension spine" FORCE)`
